### PR TITLE
Implement a basic timer class

### PR DESCRIPTION
**Pull request by F1F7Y**
Implements a basic timer class for measuring time

Example usage:
```cpp
#include "utils/timer.h"

...

CTimer timer;
timer.Start();

...

spdlog::info("Thing took {:d} ms to complete", timer.GetDuration());
```

### CODE REVIEW

**Comment by GeckoEidechse** [comment]
Out of curiosity what's reason for adding this? E.g. debugging runtime of certain functions?
(Having that kind of info in the PR description helps with better understanding the pull request and whether to add it ^^)

**Comment by F1F7Y** [comment]
We already track time intervals in some places ( [example](https://github.com/R2Northstar/NorthstarLauncher/blob/main/NorthstarDLL/engine/hoststate.cpp#L82) ). This is just a helper class so that one doesn't have to manually store the start point and then subtract to get the duration.

File: NorthstarDLL/util/timer.h [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+	void Start(void);
+	long long GetDuration(void) const;
+
+  private:
+	std::chrono::time_point<std::chrono::steady_clock> m_Clock;
+};
+
+//-----------------------------------------------------------------------------
+// Purpose:
+//-----------------------------------------------------------------------------
```
Comment (uniboi) [inline NorthstarDLL/util/timer.h:20]: mfw

Comment (F1F7Y) [reply NorthstarDLL/util/timer.h:20]: image

This is on main, unless you have a better idea and want to format all of this we stick with this :)

Comment (uniboi) [reply NorthstarDLL/util/timer.h:20]: at least write what is does exactly

**Pull request closed without merging**